src/core/store.ts

```diff
@@ -188,7 +188,7 @@ export const useStore = (toastOptions: DefaultToastOptions = {}): State => {
         listeners.splice(index, 1);
       }
     };
-  }, [state]);
+  }, []);
```

Comment:
👍 this looks good to me. 

Question: should `setState` be in the deps list instead? I don't think `useState` ever returns a different "setter" function after the first invocation but it could.

e.g.
```
  useEffect(() => {
    listeners.push(setState);
    return () => {
      const index = listeners.indexOf(setState);
      if (index > -1) {
         listeners.splice(index, 1);
       }
     };
   }, [setState]);
```

If we were using eslint we could add the `react-hooks/exhaustive-deps` rule to check for these issues.

Reply:
I'm pretty sure, `react-hooks/exhaustive-deps` won't "yell" at us for omitting `setState` from the dep array.
It's guaranteed (by React) that `setState` will be stable for the lifetime of the component.

Reply:
Ah yes you are correct:

> React guarantees that setState function identity is stable and won’t change on re-renders. This is why it’s safe to omit from the useEffect or useCallback dependency list.
> 
> https://reactjs.org/docs/hooks-reference.html#usestate